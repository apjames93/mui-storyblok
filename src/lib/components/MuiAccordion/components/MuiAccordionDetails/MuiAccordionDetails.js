import React from 'react';
import PropTypes from 'prop-types';
import { AccordionDetails } from '@material-ui/core';
import Storyblok from '../../../../utils/Storyblok';
import { validComponents } from '../../../../utils/customProps';
import MuiAccordionTypography from '../MuiAccordionTypography/MuiAccordionTypography';

const MuiAccordionDetails = ({ rootClass, content }) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AccordionDetails
      className={styles.root}
    >
      <MuiAccordionTypography {...content[0]} />
    </AccordionDetails>
  );
};

export default MuiAccordionDetails;

MuiAccordionDetails.propTypes = {
  /** storyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** MuiTypography */
  content(props, propName, componentName) {
    const components = ['MuiTypography'];
    return validComponents(props, propName, componentName, components);
  },
};

MuiAccordionDetails.defaultProps = {
  rootClass: [],
  content: [],
};
