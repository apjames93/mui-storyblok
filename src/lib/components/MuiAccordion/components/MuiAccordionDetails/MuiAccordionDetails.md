##### MuiAccordionDetails has the ability to render custom components.

MuiAccordionDetails Example:

```js
const content = [
  {
    component: 'MuiAccordionTypography',
    content: [{
      component: 'MuiText',
      content: 'Mui Accordion Details',
    }],
  },
  {
    component: 'MuiAccordionTypography',
    content: [{
      component: 'MuiIcon',
      iconName: 'google',
    }],
  },
];

<MuiAccordionDetails
  content={content}
/>
```