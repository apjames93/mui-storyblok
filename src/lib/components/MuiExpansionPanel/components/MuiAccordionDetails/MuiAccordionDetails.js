import React, { createElement } from 'react';
import PropTypes from 'prop-types';
import { AccordionDetails } from '@material-ui/core';
import Storyblok from '../../../../utils/Storyblok';
import { validComponents } from '../../../../utils/customProps';
import MuiAccordionTypography from '../MuiAccordionTypography/MuiAccordionTypography';

const MuiAccordionDetails = ({ rootClass, content }) => {
  const components = {
    MuiAccordionTypography,
  };

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <AccordionDetails
      className={styles.root}
    >
      {content.map((item, index) => createElement(
        components[item.component],
        Object.assign(item, { key: index }),
      ))}
    </AccordionDetails>
  );
};

export default MuiAccordionDetails;

MuiAccordionDetails.propTypes = {
  /** storyblok multiselect of css classes */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /** MuiTypography */
  content(props, propName, componentName) {
    const components = ['MuiTypography'];
    return validComponents(props, propName, componentName, components);
  },
};

MuiAccordionDetails.defaultProps = {
  rootClass: [],
  content: [],
};
