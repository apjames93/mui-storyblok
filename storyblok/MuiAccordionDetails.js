module.exports = {
  "name": "MuiAccordionDetails",
  "display_name": null,
  "updated_at": "2020-07-20T17:43:31.981Z",
  "schema": {
    "rootClass": {
      "type": "options"
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiAccordionTypography"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiAccordionDetails",
  "component_group_uuid": null
}