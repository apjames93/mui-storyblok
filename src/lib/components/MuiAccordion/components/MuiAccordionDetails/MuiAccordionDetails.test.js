import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiAccordionDetails from './MuiAccordionDetails';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiAccordionDetails", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiAccordionTypography", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiAccordionTypography',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiText", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiText',
        text: 'text',
      }],
    }],
  };
  const comp = shallow(<MuiAccordionDetails {...props} />);
  return { comp, props };
}

describe('<MuiAccordionDetails />', () => {
  it('renders MuiAccordionDetails', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiAccordionDetails {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
