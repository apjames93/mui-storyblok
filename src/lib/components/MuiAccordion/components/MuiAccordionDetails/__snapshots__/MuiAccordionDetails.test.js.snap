// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiAccordionDetails /> snapshot 1`] = `
<div
  className="MuiAccordionDetails-root makeStyles-root-3"
>
  <div
    className="MuiAccordionTypography storyblok--outline"
    data-blok-c="{\\"name\\": \\"MuiAccordionTypography\\", \\"space\\": \\"48408\\", \\"uid\\": \\"947be9f0-47c3-4315-a95a-550f0c560eb5\\", \\"id\\": \\"307934\\"}"
  >
    <div
      className="MuiTypography-root makeStyles-root-4 MuiTypography-body1"
    >
      <span
        className="MuiTypography-root makeStyles-root-5 MuiTypography-body1"
      >
         undefined 
      </span>
    </div>
  </div>
</div>
`;
